File: Data Structures & Algorithms/daily-temperatures/submission-1.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& trt) {
        stack<pair<int, int>> st; // temp, ind
        vector<int> res(trt.size(), 0);
        for(int i=0; i<trt.size(); i++){
            while(!st.empty() && trt[i]>st.top().first){
                res[st.top().second]=(i-st.top().second);
                st.pop();
            }
            st.push({trt[i], i});
        }
        return res;

    }
};

File: Data Structures & Algorithms/subsets-ii/submission-1.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res=[[]]
        nums.sort()
        def back(ind, ar):
            
            if(ind == len(nums)):
                # res.append(ar[:])
                return 
            ar.append(nums[ind])
            res.append(ar[:])
            back(ind+1, ar)
            while(ind < len(nums)-1):
                if nums[ind] == nums[ind+1]:
                    ind+=1
                else:
                    break
            ar.pop()
            back(ind+1, ar)
        back(0, [])
        print(nums)
        return res